src/server/async_disconnect_handler.hpp

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#ifndef _ASYNC_DISCONNECT_HANDLER_HPP_
+#define _ASYNC_DISCONNECT_HANDLER_HPP_
+
+#include <memory>
+#include <thread>
+#include <mutex>
+#include <atomic>
+#include <queue>
+#include <condition_variable>
+
+/* helper class with the task of destroying connnection handlers and, by this,
+ * also waiting for the receive and transmit threads inside the handlers to terminate
+ */
+template <typename T_CONN_HANDLER>
+class async_disconnect_handler
```

Comment:
A TODO here would be needed. Or a separate issue.
This class also needs to have a timeout functionality. Based on a system parameter, the class must wait up to a certain time interval to attempt to delete an object. If the delete is not fulfilled in that interval - being that the actual delete of an object depents of finishing some I/O rountrips to the peer server - it is a catastrophic error. (something that works similar to `pgbuf_timed_sleep`)

Reply:
I have added a new epic - http://jira.cubrid.org/browse/LETS-655 - to track this issue. It is not urgent though.